### PR TITLE
First pass

### PR DESCRIPTION
**Pull request by kas240sx89**
Creates the basic profile service with an in-memory DB

### CODE REVIEW

File: profileService/cmd/http.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	log.Fatal(http.ListenAndServe(":8080", nil))
+}
+
+//GetProfile retrieves the profile attached to the passed in "id"
+func (h *HttpEndpoints) GetProfile(w http.ResponseWriter, req *http.Request) {
+
+	id := req.FormValue("id")
+	if id == "" {
+		io.WriteString(w, "id is empty\n")
+		w.WriteHeader(400)
```
Comment (JDiPierro) [inline profileService/cmd/http.go:29]: `net/http` has [constants for the various status codes](https://golang.org/src/net/http/status.go)

File: profileService/cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package main
+
+import (
+	db2 "github.com/kas240sx89/samples/profileService/internal/db"
+	service2 "github.com/kas240sx89/samples/profileService/internal/service"
```
Comment (JDiPierro) [inline profileService/cmd/main.go]: Why's the `2` appended to all of the packages?

File: profileService/cmd/main.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+)
+
+func main() {
+
+	database := db2.NewInMemoryDB()
+	var db service2.DB
+	db = &database
```
Comment (JDiPierro) [inline profileService/cmd/main.go]: These 3 lines can probably be condensed down to 1.

File: profileService/internal/models/profile_test.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestProfileToJSON(t *testing.T) {
+	pro := new(Profile)
+	Id = "12345"
+	Email = "test@something.com"
+
+	expected := `{"id":"12345","username":"","email":"test@something.com","info":"","items":null,"LastUpdated":"0001-01-01T00:00:00Z"}`
```
Comment (JDiPierro) [inline profileService/internal/models/profile_test.go:13]: If the profile gets much bigger you might want to put this in a json file in `testdata` and load it during the test.

File: profileService/internal/service/service.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+		db: database,
+	}
+}
+
+func (s *Service) HealthCheck() string {
+	return "service is ok"
+}
+
+func (s *Service) GetProfile(id string) (*models2.Profile, error) {
+	profile, err := GetProfile(id)
```
Comment (JDiPierro) [inline profileService/internal/service/service.go]: Shouldn't these be `s.db.GetProfile(id)`?  I don't see a generic `GetProfile` func. Edit: Looks like this is unfinished.

**Review by JDiPierro — COMMENTED** [review]
Looks like it's starting off well.

File: profileService/internal/models/profile_test.go (hunk 4 above)
Comment (kas240sx89) [reply profileService/internal/models/profile_test.go:13]: ya that was the plan, so i could also test from json

File: profileService/cmd/http.go (hunk 1 above)
Comment (kas240sx89) [reply profileService/cmd/http.go:29]: yep, why do I always forget that

File: profileService/internal/service/service.go (hunk 5 above)
Comment (kas240sx89) [reply profileService/internal/service/service.go]: yes that is the way it should be, this was finished and working/ tested. not sure why the s.db. got removed or why the imports got changed to model2.

File: profileService/cmd/main.go (hunk 3 above)
Comment (kas240sx89) [reply profileService/cmd/main.go]: this actually needs to be this way. this was the same issue we had before where we needed the interface to allow pointer receivers.

**Pull request merged**